File: c++/exceptionTest.cpp
#include <iostream>
#include <exception>

using namespace std;

void test(){
	try{
		throw "test exception";
	}catch(const char* e){
		clog << e;
	}
}

void test2(){
	try{
		throw exception();
	}catch(exception e){
		clog << e.what();
	}
}

int main(){
	test2();
	return 0;
}

File: scala/exceptionTest.scala

object Test{
	def main(args: Array[String]):Unit = {
		test()
	}

	def test():Unit = {
		try{
			val a = 2/0
		}catch{
			case e:ArithmeticException => println(e)
			case e:Exception => println(e)
		}finally{
			println("apple")
		}
	}
}